api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/util/SchemaVersionChecker.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2014-2018 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.api.jaxrs.util;
+
+import org.jboss.logging.Logger;
+
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Session;
+
+import sun.misc.Version;
+
+/**
+ * @author jsanda
+ */
+public class SchemaVersionChecker {
+
+    private static final Logger logger = Logger.getLogger(Version.class);
+
+    public void waitForSchemaUpdates(Session session, String keyspace, String version, long delay, int maxRetries)
+            throws InterruptedException {
+        String configId = "org.hawkular.metrics";
+        String configName = "version";
+        int retries = 0;
+        PreparedStatement getVersion = null;
+
+        while (maxRetries > retries) {
+            try {
+                if (getVersion == null) {
+                    getVersion = session.prepare("SELECT value FROM " + keyspace + ".sys_config " +
+                            "WHERE config_id = ? AND name = ?");
+                }
+                ResultSet resultSet = session.execute(getVersion.bind(configId, configName));
+                if (!resultSet.isExhausted() && version.equals(resultSet.one().getString(0))) {
+                    logger.infof("Hawkular Metrics is up to date at version %s", version);
+                    return;
+                }
+            } catch (Exception e) {
+                if (logger.isDebugEnabled()) {
```

Comment:
.. what's the point of this? Why couldn't it just be an infof logging even if debug is enabled?

Reply:
I only log the exception stack trace with the debug statement.